### PR TITLE
Fixing up the README

### PR DESCRIPTION
**Pull request by daviddavis**
Fixing some small issues I found in the README like missing punctuation, trailing whitespace, etc.

### CODE REVIEW

**Comment by jfrazelle** [comment]
LGTM

**Comment by icecrime** [comment]
Sorry, the trailing whitespace are there for a reason (cf. #11976)! However, the other fixes look good, so please revert these changes and we'll be able to merge.

**Comment by thaJeztah** [comment]
@icecrime I quickly browsed through the whitespace changes, and I think (in this case) they are safe to remove. If I'm correct, this file is only viewed on GitHub and formatting looked good there.

**Comment by daviddavis** [comment]
I've re-added the whitespace that occurs in the middle of paragraphs (to be safe). I've removed the white space at the end of bulleted lines, paragraphs, and empty lines.

File: README.md [hunk 1]
```diff
@@ -105,7 +105,7 @@ This is usually difficult for several reasons:
     these situations with various degrees of ease - but they all
     handle them in different and incompatible ways, which again forces
     the developer to do extra work.
-  
+
```
Comment (daviddavis) [inline README.md]: Empty line here. No whitespace needed even though it's indented. There's no whitespace on line 102 either.

File: README.md [hunk 2]
```diff
@@ -213,9 +213,10 @@ We are always open to suggestions on process improvements, and are always lookin
 please see the [NOTICE](https://github.com/docker/docker/blob/master/NOTICE) document in this repo.*
 
 Use and transfer of Docker may be subject to certain restrictions by the
-United States and other governments.  
+United States and other governments.
+
```
Comment (daviddavis) [inline README.md]: Added blank line.

File: README.md [hunk 3]
```diff
@@ -213,9 +213,10 @@ We are always open to suggestions on process improvements, and are always lookin
...
 It is your responsibility to ensure that your use and/or transfer does not
-violate applicable laws. 
+violate applicable laws.
```
Comment (daviddavis) [inline README.md]: Whitespace unnecessary.

File: README.md [hunk 4]
```diff
@@ -230,14 +231,14 @@ Other Docker Related Projects
 =============================
 There are a number of projects under development that are based on Docker's
 core technology. These projects expand the tooling built around the
-Docker platform to broaden its application and utility. 
+Docker platform to broaden its application and utility.
```
Comment (daviddavis) [inline README.md]: Whitespace unnecessary.

File: README.md [hunk 5]
```diff
@@ -230,14 +231,14 @@ Other Docker Related Projects
...
 
 * [Docker Registry](https://github.com/docker/distribution): Registry 
 server for Docker (hosting/delivery of repositories and images)
 * [Docker Machine](https://github.com/docker/machine): Machine management 
-for a container-centric world 
+for a container-centric world
```
Comment (daviddavis) [inline README.md]: Whitespace unnecessary at the end of bulleted list item.

File: README.md [hunk 6]
```diff
@@ -230,14 +231,14 @@ Other Docker Related Projects
...
 * [Docker Swarm](https://github.com/docker/swarm): A Docker-native clustering 
-system 
+system
```
Comment (daviddavis) [inline README.md]: Whitespace unnecessary at the end of bulleted list item.

**Comment by daviddavis** [comment]
I annotated the whitespace removals I left. Hope that's ok.

**Comment by daviddavis** [comment]
Also, I noticed that bulleted list in [Better than VMs](https://github.com/daviddavis/docker/blob/README/README.md#better-than-vms) is of a different format than [Escape dependency hell](https://github.com/daviddavis/docker/blob/README/README.md#escape-dependency-hell). The latter has empty lines between bullet items. Want me to fix that?

**Comment by jfrazelle** [comment]
yep looks fine in the diff

**Comment by jfrazelle** [comment]
LGTM

**Comment by moxiegirl** [comment]
LGTM

**Pull request merged**